index.js

```diff
@@ -287,4 +287,73 @@ Cookies.connect = Cookies.express = function(keys) {
 
 Cookies.Cookie = Cookie
 
+function Keygrip (keys, algorithm, encoding) {
+  if (!(this instanceof Keygrip)) return new Keygrip(keys, algorithm, encoding)
+
+  if (!keys || !(0 in keys)) {
+    throw new Error('Keys must be provided.')
+  }
+
+  this.keys = keys
+  this.algorithm = algorithm || 'sha1'
+  this.encoding = encoding || 'base64'
+}
+
+Keygrip.prototype.sign = function sign (data) {
+  return hmac(this.algorithm, this.keys[0], data, this.encoding)
+}
+
+Keygrip.prototype.index = function index (data, digest) {
+  for (var i = 0, l = this.keys.length; i < l; i++) {
+    var computed = hmac(this.algorithm, this.keys[i], data, this.encoding)
+    if (constantTimeCompare(digest, computed)) {
+      return i
+    }
+  }
+
+  return -1
+}
+
+Keygrip.prototype.verify = function verify (data, digest) {
+  return this.index(data, digest) > -1
+}
+
+function hmac (algorithm, key, data, encoding) {
+  return crypto
+    .createHmac(algorithm, key)
+    .update(data)
+    .digest(encoding)
+    .replace(/\/|\+|=/g, function (x) {
+      return ({ '/': '_', '+': '-', '=': '' })[x]
+    })
+}
+
+function constantTimeCompare (a, b) {
+  var sa = String(a)
+  var sb = String(b)
+  var key = crypto.pseudoRandomBytes(32)
```

Comment:
```suggestion
  var key = crypto.randomBytes(32)
```
Maybe use `randomBytes` instead?